SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java

```diff
@@ -148,8 +151,17 @@ private boolean writeTailMetadata(boolean finished) {
     return jsonObjectFileHelper.writeObject(tailMetadata, path, log);
   }
 
+  /**
+   * Return a String for generating a PathMatcher.
+   * The matching files are caught by the S3 Uploader and pushed to S3.
+   * @return file glob String.
+   */
   private String getS3Glob() {
-    return String.format("%s*.gz*", taskDefinition.getServiceLogOutPath().getFileName());
+    List<String> fileNames = new LinkedList<>();
+    fileNames.add(taskDefinition.getServiceLogOutPath().getFileName().toString());
+    fileNames.addAll(Arrays.asList(configuration.getS3FilesToBackup()));
+
+    return String.format("{%s}*.gz*", Joiner.on(",").join(fileNames));
```

Comment:
just thinking out loud here -- i was originally going to suggest doing something like `Joiner.on(",").join(Iterables.concat(Collections.emptyList(taskDefinition.getServiceLogOutPath().getFileName().toString()), Arrays.asList(configuration.getS3FilesToBackup())))` to prevent creating an extra list here, but that's kinda gross / harder to understand